tests/cross/src/test/scala/tests/pc/CompletionOverrideAllSuite.scala

```diff
@@ -0,0 +1,180 @@
+package tests.pc
+
+import tests.BaseCompletionSuite
+
+object CompletionOverrideAllSuite extends BaseCompletionSuite {
+
+  override def beforeAll(): Unit = {
+    indexJDK()
+  }
+
+  check(
+    "simple",
+    """
+      |package example
+      |
+      |trait Foo {
+      |  def foo: Int
+      |  def bar: Int
+      |}
+      |object Main {
+      |  val x = new Foo {
+      |    def@@
+      |  }
+      |}
+      |""".stripMargin,
+    """|Implement all members(2 members)
+       |""".stripMargin,
+    filter = _.contains("Implement")
+  )
+
+  check(
+    "simple-three",
+    """
+      |package example
+      |
+      |trait Foo {
+      |  def foo: Int
+      |  def bar: Int
+      |  def car: Int
+      |}
+      |object Main {
+      |  val x = new Foo {
+      |    def@@
+      |  }
+      |}
+      |""".stripMargin,
+    """|Implement all members(3 members)
+       |""".stripMargin,
+    filter = _.contains("Implement")
+  )
+
+  check(
+    "two-left",
+    """
+      |package example
+      |
+      |trait Foo {
+      |  def foo: Int
+      |  def bar: Int
+      |  def car: Int
+      |}
+      |object Main {
+      |  val x = new Foo {
+           def foo = 2
+      |    def@@
+      |  }
+      |}
+      |""".stripMargin,
+    """|Implement all members(2 members)
+       |""".stripMargin,
+    filter = _.contains("Implement")
+  )
+
+  check(
+    "none-with-one",
+    """
+      |package example
+      |
+      |trait Foo {
+      |  def foo: Int
+      |}
+      |object Main {
+      |  val x = new Foo {
+      |    def@@
+      |  }
+      |}
+      |""".stripMargin,
+    "",
+    filter = _.contains("Implement")
+  )
+
+  check(
+    "same-scope",
+    """
+      |package a { class Foo }
+      |package b { class Foo }
+      |
+      |trait Foo {
+      |  def one: a.Foo
+      |  def two: b.Foo
+      |}
+      |object Main {
+      |  val x = new Foo {
+      |    def@@
+      |  }
+      |}
+      |""".stripMargin,
+    """|Implement all members(2 members)
```

Comment:
How does the code look like after running the completion?

Reply:
It will end up looking like this
```scala
package a { class Foo }
package b { class Foo }

trait Foo {
  def one: a.Foo
  def two: b.Foo
}
object Main {
  val x = new Foo {
    def one: a.Foo = ???
    def two: b.Foo = ???
  }
}
```

Reply:
we should have a test case to ensure it produces that code, if we render the methods separately there's a risk we produce
```
import a.Foo
import b.Foo
...
   def one: Foo = ???
   def two: Foo = ???
```

Reply:
Sounds good. I added a few more tests in for the various cases to actually check what will be produced.